### PR TITLE
[PYTHON][FASTAPI] Upgrades & improvements

### PR DESCRIPTION
**Pull request by micaelmalta**
Upgrade to Python 3.9
Upgrade all dependencies

Add nginx-unit tests
Add uvicorn tests
Add hypercorn tests
Add SQLAlchemy test

Improve PostgreSQL connexion

Benchmark results: https://www.techempower.com/benchmarks/#section=test&shareid=2096ca01-8bc7-491c-9d17-7054419c7f62&hw=ph&test=fortune&a=2

### CODE REVIEW

**Comment by micaelmalta** [comment]
I believe checks are now passing 💯

**Comment by nbrady-techempower** [comment]
Hi @micaelmalta - This has some conflicts from dependabot finding security vulns, can you update the requirements when you have a chance.

File: frameworks/Python/fastapi/app.py [hunk 1]
```diff
@@ -46,73 +53,80 @@ def get_num_queries(queries):
...
 async def single_database_query():
     row_id = randint(1, 10000)
 
-    async with connection_pool.acquire() as connection:
+    async with app.state.connection_pool.acquire() as connection:
         number = await connection.fetchval(READ_ROW_SQL, row_id)
 
-    return UJSONResponse({'id': row_id, 'randomNumber': number})
-
+    return UJSONResponse({"id": row_id, "randomNumber": number})
```
Comment (joanhey) [inline frameworks/Python/fastapi/app.py:78]: Here too.

**Review by joanhey — CHANGES_REQUESTED** [review]

**Comment by nbrady-techempower** [comment]
Going to close this for now

**Pull request closed without merging**